source/Cosmos.Core/AdvancedObject.cs

```diff
@@ -0,0 +1,53 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Cosmos.Core;
+using IL2CPU.API;
+
+namespace Cosmos.Core
+{
+    /// <summary>
+    /// This Class is created for Add more methods to objects by add this to base class
+    /// </summary>
+    public class AdvancedObject
+    {
+        public unsafe uint GetTypeId()
+        {
+            return GCImplementation.GetType(this);
+        }
+        /// <summary>
+        /// If you stuck on a problem run this function
+        /// <c>Waring: Runnig this function may get to crash</c>
+        /// </summary>
+        public void FixStackPoint()
+        {
+            if (NeedToFixStackPoint())
+            {
+                Global.mDebugger.Send("Stack is need to be fixed");
```

Comment:
Could you please explain in what case you need to use this function?

Reply:
I used it for to check if something is worng with the stack register, For example: Type id is not match to type id that is nedded to run the method, I saw that the proplem happened while the CallVirt op code is runnig